pkg/sfu/buffer/buffer.go

```diff
@@ -321,6 +322,9 @@ func (b *Buffer) Close() error {
 		if b.rtpStats != nil {
 			b.rtpStats.Stop()
 			b.logger.Infow("rtp stats", "direction", "upstream", "stats", b.rtpStats.ToString())
+			if b.onFinalRtpStats != nil {
```

Comment:
switched to using a callback when final stats are completed.

without this, it was tricky with timing since receivers are being cleared for multiple reasons from diff entry points.